specs/lib/writeBuffersSpec.js

```diff
@@ -39,27 +41,39 @@ describe('writeBuffers', function() {
 
     it('writes an external buffer', function(done) {
         var gltf = clone(testGltf);
+        var options = {
+            outputPath : outputPath,
+            embed : false,
+            embedImage : false,
+            createDirectory : true
+        };
 
-        writeGltf(gltf, outputPath, false, false, true, function() {
-            expect(gltf.buffers.CesiumTexturedBoxTest.extras).not.toBeDefined();
-            expect(gltf.buffers.CesiumTexturedBoxTest.uri).toEqual('CesiumTexturedBoxTest.bin');
-            fs.readFile(outputBufferPath, function(err, outputData) {
-                if (err) {
-                    throw err;
-                }
-                expect(bufferEqual(outputData, bufferData)).toBe(true);
-                done();
-            });
+        writeGltf(gltf, options)
+            .then(function() {
+                expect(gltf.buffers.CesiumTexturedBoxTest.extras).not.toBeDefined();
+                expect(gltf.buffers.CesiumTexturedBoxTest.uri).toEqual('CesiumTexturedBoxTest.bin');
+                readFile(outputBufferPath)
+                    .then(function(outputData) {
+                        expect(bufferEqual(outputData, bufferData)).toBe(true);

```

Comment:
I am fairly certain I shouldn't have to be nesting `.then` like this, but I am unsure how I could write this in order to have `outputData` accessible by the second `.then` function


I have attempted the code below, but that results in `outputData` being undefined.
```
writeGltf(gltf, options)
            .then(function() {
                expect(gltf.buffers.CesiumTexturedBoxTest.extras).not.toBeDefined();
                expect(gltf.buffers.CesiumTexturedBoxTest.uri).toEqual('CesiumTexturedBoxTest.bin');
            })
            .then(function() {
                readFile(outputBufferPath);
            })
            .then(function(outputData) {
                expect(bufferEqual(outputData, bufferData)).toBe(true);
                done();
            });
```